slamd/discovery/processing/experiment/models/tuned_gaussian_process_regressor.py

```diff
@@ -0,0 +1,63 @@
+from mlxtend.feature_selection import SequentialFeatureSelector
+from sklearn.gaussian_process import GaussianProcessRegressor
+from sklearn.gaussian_process.kernels import RBF, ConstantKernel
+from sklearn.model_selection import GridSearchCV
+from sklearn.pipeline import Pipeline
+
+
+class TunedGaussianProcessRegressor:
+
+    @classmethod
+    def find_best_model(cls, training_rows, training_labels):
+        # Perform a forward sequential feature selection with 3 features without cross-validation
+        sfs_gr_testing = SequentialFeatureSelector(estimator=cls._create_optimized_gaussian_process(),
+                                                   k_features=3,
+                                                   forward=True,
+                                                   floating=False,
+                                                   scoring='r2',
+                                                   cv=None)
+        pipe = Pipeline([('sfs', sfs_gr_testing),
+                         ('gp2', cls._create_optimized_gaussian_process())])
+        grid_search_cv = GridSearchCV(estimator=pipe,
+                                      param_grid=cls._create_parameters_for_grid_search(),
+                                      scoring='r2',
+                                      n_jobs=-1,
+                                      cv=4,
+                                      refit=False)
+        grid_search_cv = grid_search_cv.fit(training_rows, training_labels)
+        # Return the best model found
+        return pipe.set_params(**grid_search_cv.best_params_)
+
+    @classmethod
+    def _create_optimized_gaussian_process(cls):
```

Comment:
maybe we can find a better name. This Gauss is not yet optimized, but the chosen fixed parameters rather serves as a sensible starting point for the subsequent tuning. Maybe somethin like _create_baseline_gauss_for_further_tuning?